{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "9ed3e340-17fd-4b71-a98e-c776aa45d053",
   "metadata": {},
   "source": [
    "<!-- DATA PROVIDER INSTRUCTIONS\n",
    "\n",
    "1. Provide the name of your dataset, replacing the bracketed placeholder text.\n",
    "2. Update the Registry of Open Data landing page URL, by replacing the bracketed placeholder text. The [REGISTRY_YAML_NAME] will correspond to the name of the YAML document in your pull request to the Registry of Open Data on Github, minus the .yaml file extension.\n",
    "3. Remove these comment blocks when you have completed each section.\n",
    "\n",
    "DATA PROVIDER INSTRUCTIONS -->\n",
    "\n",
    "# Get to Know a Dataset: [JENGER]\n",
    "\n",
    "This notebook serves as a guided tour of the [[JENGER]](https://registry.opendata.aws/[jenger.yaml]) dataset. More usage examples, tutorials, and documentation for this dataset and others can be found at the [Registry of Open Data on AWS](https://registry.opendata.aws/)."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a3779654-eeee-4708-83cf-245e03303475",
   "metadata": {},
   "source": [
    "<!-- DATA PROVIDER INSTRUCTIONS\n",
    "\n",
    "The goal of this section is to orient users to the structure of your dataset. \n",
    "\n",
    "1. How are key prefixes and objects organized in your S3 bucket?\n",
    "2. What kinds of filetypes are represented in your dataset?\n",
    "3. Explain with text what users are expected to encounter, and then demonstrate with code the organizational framework you applied when creating your dataset.\n",
    "4. The responses to each question section are meant to be expanded or replaced as dictated by your dataset\n",
    "\n",
    "DATA PROVIDER INSTRUCTIONS -->\n",
    "\n",
    "### Q: How have you organized your dataset? Help us understand the key prefix structure of your S3 bucket.\n",
    "\n",
    "We have organized the dataset to allow users to easily distinguish between summary statistics and other analysis data from the JENGER portal site. specifically, we use separate top-level prefixes (e.g., Analysis Result, Other Data) for each data type.\n",
    "\n",
    " Full documentation for this dataset can be found at: http://jenger.riken.jp/en/\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dd7f4bcf-ec40-432f-a31f-4477efa205ee",
   "metadata": {
    "tags": []
   },
   "source": [
    "<!-- DATA PROVIDER INSTRUCTIONS\n",
    "This section is meant to orient users of your dataset to the formats present in your dataset, particularly if your dataset includes formats that may be unfamiliar to a general data scientist audience. This section should include:\n",
    "\n",
    "1. Explanation of data format(s) (very common formats can be very briefly described, while less common\n",
    "   or domain specific formats should include more explanation as well as links to official documentation)\n",
    "2. Explanation of why the data format was chosen for your dataset\n",
    "3. Recommendations around software and tooling to work with this data format\n",
    "4. Explanation of any dataset-specific aspects to your usage of the format\n",
    "5. Description of AWS services that may be useful to users working with your data\n",
    "DATA PROVIDER INSTRUCTIONS -->\n",
    "\n",
    "### Q: What data formats are present in your dataset? What kinds of data are stored using these formats? Can you give any advice for how you work with these data formats?\n",
    "\n",
    "Our dataset consists of:\n",
    "\n",
    "- Genome-wide association study (GWAS) summary statistics (Tab-Separated Values; flat file or gzip format)\n",
    "- Variant call format (VCF) file (bgzip + tabix format)\n",
    "- Allele frequencies of the variants identified by WGS of the Japanese population (gzip format)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7362bd15",
   "metadata": {},
   "source": [
    "<!-- DATA PROVIDER INSTRUCTIONS\n",
    "The goal of this section is to demonstrate loading a portion of data from your dataset, and reveal something about its structure.\n",
    "1. Load an object from S3\n",
    "2. Show the structure of data in the object\n",
    "DATA PROVIDER INSTRUCTIONS -->\n",
    "\n",
    "### Q: Can you show us an example of downloading and loading data from your dataset?\n",
    "\n",
    "You can browse and download the data stored in S3 through our web portal: [JENGER](http://jenger.riken.jp/en/)\n",
    "\n",
    "1. [GWAS sumstats](http://jenger.riken.jp/en/result)\n",
    "2. [Allele frequencies of the variants identified by WGS of the Japanese population](http://jenger.riken.jp/en/data)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5717987e-c9c9-4c0f-9f2e-a8d761492625",
   "metadata": {},
   "source": [
    "<!-- DATA PROVIDER INSTRUCTIONS\n",
    "The goal here is to visualize some aspect of your dataset in order to help users understand it. In addition to helping users of your dataset understand the dataset, an additional goal is to impress!\n",
    "\n",
    "Please demonstrate any data preprocessing or reshaping required for your visualization(s).\n",
    "\n",
    "https://www.reddit.com/r/dataisbeautiful/ for inspiration.\n",
    "DATA PROVIDER INSTRUCTIONS -->\n",
    "\n",
    "### Q: A picture is worth a thousand words. Show us a visual (or several!) from your dataset that either illustrates something informative about your dataset, or that you think might excite someone to dig in further.\n",
    "\n",
    "Visualizing the genetic landscape of complex traits in the Japanese population.\n",
    "\n",
    "The Manhattan plot below illustrates the genome-wide association results for Vasospastic Angina. Each dot represents a genetic variant, with the x-axis showing genomic position and the y-axis showing association strength $−log_{10}(P)$.\n",
    "\n",
    "The peaks clearly identify significant genomic loci, demonstrating the high quality of the summary statistics available in this dataset. Notably, you can observe strong signals at known loci as well as signals specific to the East Asian population.\n",
    "\n",
    "<img src=\"images/VSA.Fig1A.jpg\" width=\"30%\">\n",
    "\n",
    "[Planned Visualization Approach]\n",
    "• Data Source: Load summary statistics for VSA from s3.\n",
    "• Processing: Filter for variants with P-value < 0.01 to optimize plotting performance.\n",
    "• Plotting: Generate a scatter plot using Python (matplotlib/seaborn) where:\n",
    " - X-axis: Chromosome and Position\n",
    " - Y-axis: $−log_{10}(P)$\n",
    "• Annotation: Highlight the top significant peak with the gene name."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "183c8b85-ed1c-4f2c-bd0e-fbfbc67c4723",
   "metadata": {},
   "source": [
    "<!-- DATA PROVIDER INSTRUCTIONS\n",
    "This section is less prescriptive / freeform than previous sections. The goal here is to show an opinionated example of answering a question using your data. The scale of your dataset may preclude a full example, and so feel free to limit the scope of this example (e.g. work on a subset of data). Users should be able to replicate your example in this notebook, and get a sense of how they would scale up.\n",
    "\n",
    "A \"toy\" example is better than no example.\n",
    "\n",
    "Ideally, your example would:\n",
    "1. Transmit some of your domain & dataset experience to the reader, drawing on your own work as much as possible\n",
    "2. Provide a jumping off point for users to extend your work, and do novel work of their own.\n",
    "\n",
    "DATA PROVIDER INSTRUCTIONS -->\n",
    "\n",
    "### Q: What is one question that you have answered using these data? Can you show us how you came to that answer?\n",
    "\n",
    "Question:\n",
    "- Which genetic variants are significantly associated with Height in the Japanese population?\n",
    "\n",
    "Answer / Demonstration:\n",
    "\n",
    "- We utilized the GWAS summary statistics for Height contained in this dataset to identify significant associations. By filtering variants with a p-value less than the genome-wide significance threshold (5×10−8), we successfully replicated known risk loci and identified novel candidate variants specific to the East Asian population."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cf645724-3108-4ada-a832-10b3431eb8e2",
   "metadata": {},
   "source": [
    "<!-- DATA PROVIDER INSTRUCTIONS\n",
    "This section is, like the previous one, intended to be freeform / non-prescriptive. The goal here is to provide a challenge to the community to do something novel with your dataset. That can either be novel in terms of the task, or novel in terms of methodological or computational approach.\n",
    "\n",
    "Another way to consider this section, is as a wishlist. If you were less constrained by time, cost, skill, etc., what would you like to see achieved using these data? \n",
    "\n",
    "The challenge should, however, be somewhat realistic. A challenge that assumes e.g. original data collection, is likely to go unanswered.\n",
    "DATA PROVIDER INSTRUCTIONS -->\n",
    "\n",
    "### Q: What is one unanswered question that you think could be answered using these data? Do you have any recommendations or advice for someone wanting to answer this question?\n",
    "\n",
    "Question:\n",
    "- Can we pinpoint the causal variants driving the GWAS signals for specific diseases within the Japanese population?\n",
    "\n",
    "Recommendations or Advice:\n",
    "- While GWAS identifies associated genomic regions, identifying the exact causal variants remains a challenge due to LD. This dataset's VCF files provide the dense genotype information necessary for fine-mapping.\n",
    "- We recommend integrating this data with functional genomics annotations (such as eQTL data) and using Bayesian fine-mapping tools (like FINEMAP or SuSiE). Focus on loci where the lead variant in the Japanese population differs from that in European studies, as these regions offer the best opportunity to disentangle causal mechanisms."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "466ca021",
   "metadata": {},
   "source": [
    "# DATA PROVIDER: PLEASE REMEMBER TO CLEAR ALL OUTPUTS BEFORE COMMITTING TO YOUR GITHUB REPOSITORY"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
